### PR TITLE
[Frontend] Normalize Responses API input for multi-turn conversations

### PR DESCRIPTION
**Pull request by daniel-salib**
## Purpose

Normalizes input types in the Responses API for **Codex CLI** multi-turn conversation support.

When Codex echoes back previous responses in multi-turn conversations, it includes the exact types that were returned by the server. However, the server expects different types on input vs output:
- Server returns: `output_text` → Client should send: `input_text`
- Server returns: `output_text` in reasoning → Client should send: `reasoning_text`

This PR adds preprocessing to automatically normalize these types before validation, enabling seamless multi-turn conversations with Codex without requiring manual type conversion.

**Changes:**
- Rename `function_call_parsing` → `preprocess_input` (more descriptive)
- Add `output_text` → `input_text` conversion for message content
- Add `output_text`/`input_text` → `reasoning_text` conversion for reasoning content
- Handle `ValidatorIterator` from Pydantic that needs conversion to list
- Extract `id` from `encrypted_content` for reasoning items if missing at top level

## Test Plan

```bash
# This curl sends output_text (what the server returns) in a multi-turn conversation.
# On main: fails validation because output_text is not a valid input type
# On this branch: succeeds because output_text is normalized to input_text

curl -s http://localhost:8000/v1/responses \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer test" \
  -d '{
    "model": "gpt-oss",
    "input": [
      {
        "type": "message",
        "role": "user",
        "content": [{"type": "input_text", "text": "What is 2+2?"}]
      },
      {
        "type": "message",
        "role": "assistant",
        "content": [{"type": "output_text", "text": "4"}]
      },
      {
        "type": "message",
        "role": "user",
        "content": [{"type": "input_text", "text": "And 3+3?"}]
      }
    ]
  }'
```

## Test Result

**On main (fails):**
```json
{
  "error": {
    "message": "Invalid content type 'output_text' in message content",
    "type": "invalid_request_error"
  }
}
```

**On this branch (succeeds):**
```json
{"id":"resp_9e1f9966a958232f","created_at":1768299126,"incomplete_details":null,"instructions":null,"metadata":null,"model":"gpt-oss","object":"response","output":[{"id":"rs_a0618b7cee420419","summary":[],"type":"reasoning","content":[{"text":"User asks simple math: 3+3 = 6. Provide answer.","type":"reasoning_text"}],"encrypted_content":null,"status":null},{"id":"msg_8151065583c2fbc7","content":[{"annotations":[],"text":"6","type":"output_text","logprobs":null}],"role":"assistant","status":"completed","type":"message"}],"parallel_tool_calls":true,"temperature":1.0,"tool_choice":"auto","tools":[],"top_p":1.0,"background":false,"max_output_tokens":130983,"max_tool_calls":null,"previous_response_id":null,"prompt":null,"reasoning":null,"service_tier":"auto","status":"completed","text":null,"top_logprobs":null,"truncation":"disabled","usage":{"input_tokens":89,"input_tokens_details":{"cached_tokens":0,"input_tokens_per_turn":[89],"cached_tokens_per_turn":[0]},"output_tokens":27,"output_tokens_details":{"reasoning_tokens":17,"tool_output_tokens":0,"output_tokens_per_turn":[27],"tool_output_tokens_per_turn":[0]},"total_tokens":116},"user":null,"input_messages":null,"output_messages":null}
```

### CODE REVIEW

**Review by qandrew — CHANGES_REQUESTED** [review]
ResponsesAPI doesn't seem to allow input_text / output_text, the types should be `message`. I don't think we should modify our responsesAPI in vLLM to support that; the client can have some code to change the type to `message` ?

https://platform.openai.com/docs/api-reference/responses/create#responses_create-input-input_item_list-input_message

File: vllm/entrypoints/openai/protocol.py [hunk 1]
```diff
@@ -298,6 +298,47 @@ def get_logits_processors(
...
+                    item["id"] = encrypted["id"]
+            except (json.JSONDecodeError, TypeError):
+                pass
+
+        # Convert output_text/input_text to reasoning_text
+        if isinstance(content, list):
+            for content_item in content:
+                if isinstance(content_item, dict):
+                    ctype = content_item.get("type")
+                    if ctype in ("output_text", "input_text"):
```
Comment (qandrew) [inline vllm/entrypoints/openai/protocol.py:338]: i don't think this should be allowed as it's not specified in responsesAPI?

**Comment by daniel-salib** [comment]
opting for client side handling
